test/fn-api-tests/calls_test.go

```diff
@@ -42,15 +44,15 @@ func TestCallsDummy(t *testing.T) {
 		t.Error("Must fail because `dummy` call does not exist.")
 	}
 
-	DeleteApp(t, s.Context, s.Client, s.AppName)
```

Comment:
The reason to have app provisioning and deleting on every test was to make tests independent, to make them be capable to be executed completely independently from any other test. However, this change brings the certain inconsistency between tests, so they are dependent.

Reply:
A agree with the above -  that's why I added `defer s.Cleanup()` -  now deletes any apps that were created within the test (even on a panic) - so tests are independent. .

Reply:
I've rejigged the purpose of `SetupSuite` a bit - (hence rename to `TestHarness`) - it's now a per-test case harness

Reply:
Noticed an issue where I wasn't properly cleaning up apps in this case, have fixed.